### PR TITLE
Updated UI, fixed bugs and refactored code

### PR DESCRIPTION
**Pull request by ItsAdityaKSingh**
## This PR fixes issues:
- #22 
- #23 
- #25 
- #32 
- #34 
- #44 

Along with the **UI improvements** and minor **bug** fixes, I have minutely **refactored** the code for proper readability and understanding, added **commas(,)** at a few places, fixed **sizing issues** for all the containers and buttons, and also tried to improve the **overall interface**. Any suggestions for improvements are most welcome!

### CODE REVIEW

**Comment by ItsAdityaKSingh** [comment]
Fixed the issue #44  here itself

**Review by nb9960 — CHANGES_REQUESTED** [review]

File: lib/components/create_join_dialog.dart [hunk 1]
```diff
@@ -7,174 +7,189 @@ import 'package:beacon/view_model/home_view_model.dart';
...
+                            model.durationController.text = model
+                                .resultingDuration
+                                .toString()
+                                .substring(0, 8);
+                          },
+                          validator: (value) {
+                            if (value.startsWith("0:00:00"))
+                              return "Enter valid duration";
+                            return null;
+                          },
```
[the remark this answers is not in the document]
Comment (ItsAdityaKSingh) [reply lib/components/create_join_dialog.dart]: Yes, my bad, they were changed while removing some of my bug-fixing commits earlier😅. I have fixed them! Really sorry for the trouble caused.

**Comment by nb9960** [comment]
Hey @ItsAdityaKSingh, This PR still doesn't reflect the desired changes. In fact, UI had no issues earlier, but now it has (adding the screenshot below for reference). Maybe run gh pr checkout 45 or make sure you've pulled the changes locally and then try to test your changes.

**Review by nb9960 — COMMENTED** [review]
Hey @ItsAdityaKSingh, This PR still doesn't reflect the desired changes. In fact, UI had no issues earlier, but now it has (adding the screenshot below for reference). Maybe run gh pr checkout 45 or make sure you've pulled the changes locally and then try to test your changes.

**Review by nb9960 — APPROVED** [review]

**Comment by nb9960** [comment]
Thanks for the PR @ItsAdityaKSingh , Merged! :tada:

**Pull request merged**